api/src/main/java/org/apache/iceberg/expressions/UnboundPredicate.java

```diff
@@ -125,13 +132,26 @@ public Expression bind(Types.StructType struct, boolean caseSensitive) {
         case LT_EQ:
         case EQ:
           return Expressions.alwaysFalse();
-//        case IN:
-//          break;
-//        case NOT_IN:
-//          break;
+        case IN:
+        case NOT_IN:
+          break;
       }
     }
-    return new BoundPredicate<>(op(), new BoundReference<>(field.fieldId(),
-        schema.accessorForField(field.fieldId())), lit);
+
+    switch (op()) {
+      case IN:
+      case NOT_IN:
+        @SuppressWarnings("unchecked")
+        Set<Literal<T>> lits = literalSet()
+                .stream()
+                .map(l -> (Literal<T>) l.to(field.type()))
+                .filter(Objects::nonNull)
```

Comment:
This filter changes the predicate. If any object is null, then the value could not be converted to the right type and a validation exception should be thrown, like the case above for predicates with a single literal.

Reply:
Thanks for the comments. Fixed it.